week4/week4Pllayground.playground/Contents.swift

```diff
@@ -0,0 +1,241 @@
+//Call the function twice, sending a different Animal each time.
+
+
+enum Animals {
+  case kobe
+  case chronic
+  case grey
+  case blanketMonsater
+  case filamentDog
+}
+
+func theSoundMadeBy (animal: Animals) {
+  switch animal {
+  case .kobe:
+    print("Kobe goes woo woo woo")
+  case .chronic:
+    print("Chronic goes grrr......")
+  case .grey:
+    print("Grey goes gr... gr... gr..")
+  case .blanketMonsater:
+    print("Blanket Monster is gonna get you")
+  case .filamentDog:
+    print("Fliament dog goes bzzz.......bzz...")
+  }
+}
+
+theSoundMadeBy(animal: .kobe)
+theSoundMadeBy(animal: .chronic)
+
+//This question will have you creating multiple functions that will require you to use closures and collections. First, you will do some setup.
+//Create an array of Int called nums with the values of 0 to 100.
+//Create an array of Int? called numsWithNil with the following values:
+//79, nil, 80, nil, 90, nil, 100, 72
+//Create an array of Int called numsBy2 with values starting at 2 through 100, by 2.
+//Create an array of Int called numsBy4 with values starting at 2 through 100, by 4.
+
+var nums = Array(1...100)
+var numsWithNil: [Int?] = [79, nil, 80, nil, 90, nil, 100, 72]
+
+var numsBy2: [Int] = []
+for num in stride(from: 2, through: 100, by: 2) {
+  numsBy2.append(num)
+}
+
+var numsBy4: [Int] = []
+for num in stride (from: 2, through: 100, by: 4) {
+  numsBy4.append(num)
+}
+
+//- Create a function called evenNumbersArray that takes a parameter of [Int] (array of Int) and returns [Int]. The array of Int returned should contain all the even numbers in the array passed. Call the function passing the nums array and print the output.
+func evenNumbersArray (arrayOfInt:[Int]) -> [Int] {
+  var evenNumbers: [Int] = []
+  for int in arrayOfInt {
+    if int % 2 == 0 {
+      evenNumbers.append(int)
+    }
+  }
+  return evenNumbers
+}
+
+var evenNums = evenNumbersArray(arrayOfInt: nums)
+print(evenNums)
+
+// Create a function called sumOfArray that takes a parameter of [Int?] and returns an Int. The function should return the sum of the array values passed that are not nil. Call the function passing the numsWithNil array, and print out the results.
+
+func sumOfArray (num: [Int?]) -> Int {
+  var sum = 0
+  for num in num {
+    if let unwrappedNum = num {
+      sum += unwrappedNum
+    }
+  }
+  return sum
+  }
+
+sumOfArray(num: numsWithNil)
+
+////- Create a function called commonElementsSet that takes two parameters of [Int] and returns a Set<Int> (set of Int.) The function will return a Set<Int> of the values in both arrays.
+//Call the function commonElementsSet passing the arrays numsBy2, numsBy4, and print out the results.
+
+func commonElementsSet(array1: [Int], array2: [Int]) -> Set<Int> {
+  let array1 = Set(array1)
+  let array2 = Set(array2)
+  
+  let commonSet = array1.intersection(array2)
+  
+  return commonSet
+}
```

Comment:
Nice. My only note here is that `array1` and `array2` are not actually arrays but Sets.